packages/ketcher-core/src/application/render/restruct/restruct.ts

```diff
@@ -158,6 +157,22 @@ class ReStruct {
     });
   }
 
+  get visibleRGroupAttachmentPoints() {
+    const sgroups = this.molecule.sgroups;
+    const functionalGroups = this.molecule.functionalGroups;
+    return this.rgroupAttachmentPoints.filter((_id, reItem) => {
+      const atomId = reItem.item.atomId;
+      const atom = this.molecule.atoms.get(atomId);
+      assert(atom != null);
+      return !FunctionalGroup.isAtomInContractedFunctionalGroup(
```

Comment:
Sorry, I think I don't really get your question, but because this function is for getting R-Group attachment points that are outside of contracted functional groups, does it make sense?